multiqc/modules/sargasso/sargasso.py

```diff
@@ -0,0 +1,154 @@
+#!/usr/bin/env python
+
+""" MultiQC module to parse output from sargasso """
+
+from __future__ import print_function
+from collections import OrderedDict
+import logging
+
+from multiqc import config
+from multiqc.plots import bargraph
+from multiqc.modules.base_module import BaseMultiqcModule
+
+# Initialise the logger
+log = logging.getLogger(__name__)
+
+class MultiqcModule(BaseMultiqcModule):
+
+    def __init__(self):
+
+        # Initialise the parent object
+        super(MultiqcModule, self).__init__(name='sargasso',
+        anchor='sargasso',
+        href='http://statbio.github.io/Sargasso/',
+        info="is a tool to separate mixed-species RNA-seq reads"\
+             "according to their species of origin.")
+
+         # Find and load any Sargasso reports
+        self.sargasso_data = dict()
+        self.sargasso_files = list()
+        self.sargasso_keys = list() #header keys
+
+        for f in self.find_log_files('sargasso'):
+            self.parse_sargasso_logs(f)
+            self.sargasso_files.append(f)
+
+        # log.info('Removing ignored samples...')
+        self.sargasso_data = self.ignore_samples(self.sargasso_data)
+
+        if len(self.sargasso_data) == 0:
+            raise UserWarning
+
+        log.info("Found {} reports".format(len(self.sargasso_files)))
+
+        # Write parsed report data to a file
+        self.write_data_file(self.sargasso_data, 'multiqc_sargasso')
+
+        # Basic Stats Table
+        self.sargasso_stats_table()
+
+        # Assignment bar plot
+        self.add_section( plot = self.sargasso_chart() )
+
+        # log.info('done')
+
+    def parse_sargasso_logs(self, f):
+        """ Parse the sargasso log file. """
+        species_name = list()
+        items = list()
+        header = list()
+        for l in f['f'].splitlines():
+            s = l.split(",")
+            if len(s) < 7:
+                continue
```

Comment:
If you think that this may be a risk, it can be nice to add a counter to make sure that we don't spend ages reading through millions of lines of an incorrect file. Very minor thing though, I don't think that it's a high risk given the specific filename of the logs etc.